### PR TITLE
Return all

### PR DESCRIPTION
**Pull request by attiks**
Show all defined factoids

### CODE REVIEW

File: src/factoids.coffee [hunk 1]
```diff
@@ -76,8 +77,15 @@ module.exports = (robot) ->
       msg.reply 'Not a factoid'
 
   robot.respond /factoids?/i, (msg) =>
-    url = process.env.HUBOT_BASE_URL or "http://not-yet-set/"
-    msg.reply "#{url.replace /\/$/, ''}/#{robot.name}/factoids"
+    all = @factoids.getAll()
```
Comment (therealklanni) [inline src/factoids.coffee]: Rather than replacing the functionality of this command, let's add a separate (new) command.

**Review by therealklanni — CHANGES_REQUESTED** [review]

File: src/factoids.coffee (hunk 1 above)
Comment (attiks) [inline src/factoids.coffee]: Any preference for the name of the new command?

Comment (therealklanni) [inline src/factoids.coffee]: No preference, as long as it's intuitive. 👍

**Comment by attiks** [comment]
Finally found to the time, renamed to *List all factoids*

**Comment by attiks** [comment]
Let me know if you want some more changes, but I think it's good to go

File: test/factoids_test.coffee [hunk 2]
```diff
@@ -144,12 +145,24 @@ describe 'factoids', ->
...
+      robot.brain.data.factoids.bas = value: 'baz', forgotten: false
+
+    it 'responds to list all factoids', (done) ->
+      adapter.on 'reply', (envelope, strings) ->
+        expect(strings[0]).to.match /All factoids: \n!bas: baz\n/
+        done()
+
+      adapter.receive(new TextMessage user, 'hubot: list all factoids')
+
+  it 'responds to list all factoids', (done) ->
```
Comment (therealklanni) [inline test/factoids_test.coffee]: Looks like L160-165 needs to be reverted?

**Review by therealklanni — CHANGES_REQUESTED** [review]

File: test/factoids_test.coffee (hunk 2 above)
Comment (attiks) [inline test/factoids_test.coffee]: It's to test the behavior if there's no factoid defined

Comment (therealklanni) [inline test/factoids_test.coffee]: That's fine, if you want to add it, but it looks like you removed the test for the URL?

**Review by therealklanni — APPROVED** [review]
lgtm. Thanks for your efforts!

I will get this merged and published later today.

**Comment by therealklanni** [comment]
Sorry for the wait! (Completely spaced it)

hubot-factoids 1.2.0 published 🎉

**Pull request merged**